wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.swift

```diff
@@ -500,6 +500,43 @@ extension ConversationContentViewController: UITableViewDelegate {
     func tableView(_ tableView: UITableView, willSelectRowAt indexPath: IndexPath) -> IndexPath? {
         willSelectRow(at: indexPath, tableView: tableView)
     }
+
+    func tableView(
+        _ tableView: UITableView,
+        leadingSwipeActionsConfigurationForRowAt indexPath: IndexPath
+    ) -> UISwipeActionsConfiguration? {
+
+        let sections = dataSource.currentSections
+        guard
+            sections.indices.contains(indexPath.section),
+            sections[indexPath.section].elements.indices.contains(indexPath.row),
+            sections[indexPath.section].elements[indexPath.row].instance.supportsActions,
+            let actionController = sections[indexPath.section].elements[indexPath.row].actionController,
+            actionController.canPerformAction(action: .reply)
+        else { return nil }
```

Comment:
question: could you explain what's the logic here? is it still 1 section is 1 message?

Reply:
It's basically just about calling `supportsActions` and `canPerformAction(action: .reply)`.